### DISCUSSION TITLE
How to declare Multiple type for request body?

### DISCUSSION THREAD

**Discussion by jet10000** [fastapi/fastapi#7175] [Questions] [answered]
```
class A:
class B:
class C:

@app.put("/test")
async def test(item: OptionalA]):
    return item
```

I want to allow class A,B,C for test path.

**Comment by ycd** [comment, marked as answer]
I did not understand the problem here, but as far as I understand from the title you want to do this.

```python
from fastapi import FastAPI, Body
from typing import Union
from pydantic import BaseModel


class User(BaseModel):
    name: str


class Item(BaseModel):
    size: int
    price: float


app = FastAPI()


@app.post("/multi/")
def process_things(body: Union[User, Item] = Body(..., example={"Create your own schema.": "here"})):
    return body
```

The reason that I used `Body(..., example="here")` is #1083, this is not a bug in FastAPI but Swagger can not document Union models properly right now, so you should create your own example if needed.

**Comment by jet10000** [comment]
Yes, Thanks

**Comment by tiangolo** [comment]
Thanks for the help @ycd! 🍰 

And thanks @jet10000 for closing the issue

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
